contributors/design-proposals/admission_control_extension.md

```diff
@@ -0,0 +1,507 @@
+    Configuration would be similar to registering new API group versions and include config like "fail open" or
+    "fail closed".
+
+Some admission controller types would not be possible for these extensions:
+
+* Mutating admission webhooks could be a later addition
+* Admission controllers that need access to the acting user can receive that via the external webhook.
+* Admission controllers that "react" to the acting user can couple the information received via a webhook and then act if they observe mutation succeed (tuple combining resource UID and resource generation).
+* Quota will continue to be a core plugin per API server, so extension is not critical.
+
+
+
+#### Implications:
+
+1. Initializers and generic admission controllers are highly privileged, so while some separation is valuable they are effectively cluster scoped
+2. This mechanism would allow dedicated infrastructure to host admission for multiple clusters, and allow some expensive admission to be centralized (like quota which is hard to performantly distribute)
+3. There is no way to build initializers for updates without a much more complicated model, but we anticipate initializers to work best on creation.
+4. Ordering will probably be necessary on initializers because defaulting in the wild requires ordering. Non-mutating validation on the other hand can be fully parallel.
+5. Some admission depends on knowing the identity of the actor - we will likely need to include the **creator** as information to initializers.
+6. Quota must still run after all validators are invoked. We may need to make quota extensible in the future.
+
+
+#### Initializers
+
+An initializer must allow some API clients to perceive creations prior to other apps. For backwards compatibility,
+uninitialized objects must be invisible to legacy clients. In addition, initializers must be recognized as participating
+in the initialization process and therefore a list of initializers must be populated onto each new object. Like
+finalizers, each initializer should perform an update and remove itself from the object.
+
+Every API object will have the following field added:
+
+```
+type ObjectMeta struct {
+    ...
+    // Initializers is a list of initializers that must run prior to this object being visible to
+    // normal clients. Initializers are executed in order. This field is nil if initialization is not
+    // specified, or empty if no initializers should be run. Only highly privileged users
+    // may modify this field.
+    Initializers []Initializer `json:"initializers"`
+    ...
+}
+
+type Initializer struct {
+    // Name is the name of the process that owns this initializer
+    Name string `json:"name"`
+    // Error is set if the initializer cannot make progress
+    Status *metav1.Status `json:"status"`
+}
+```
+
+On creation, an admission controller defaults the initializers field (if nil) to a value from the system
```

Comment:
I think I disagree with you, as evidenced by that thread.  JSON makes the distinction, we have support for it (except in a couple of cases).  We can queue that up for a discussion in API machinery.